lib/ansible/module_utils/redfish_utils.py

```diff
@@ -128,10 +128,14 @@ def _find_systems_resource(self, uri):
             if response['ret'] is False:
                 return response
             data = response['data']
-            for member in data[u'Members']:
-                systems_service = member[u'@odata.id']
-            self.systems_uri = systems_service
-            return {'ret': True}
+            if data.get(u'Members'):
+                for member in data[u'Members']:
+                    systems_service = member[u'@odata.id']
+                    self.systems_uri = systems_service
+                    return {'ret': True}
```

Comment:
This construction seems a bit awkward as it's using a loop that will only happen once when no loop is needed.  Probably better to rewrite so it's clear what is really being looked for:
``` python
            if data.get(u'Members'):
                systems_service = data[u'Members'][0][u'@odata.id']
                self.systems_uri = systems_service
                return {'ret': True}
```

Since this is fixing a bug and has already been merged to devel, I'm merging it but please consider updating the code so it's easier to work on it in the future.

Reply:
@abadger got it. thanks :)